pkg/runtime/runtime.go

```diff
@@ -1878,6 +1913,272 @@ func (a *DaprRuntime) initNameResolution() error {
 	return nil
 }
 
+func (a *DaprRuntime) sendBulkToDeadLetter(
+	name string, msg *pubsub.BulkMessage, deadLetterTopic string,
+	bulkResponses []pubsub.BulkSubscribeResponseEntry,
+) error {
+	data := make([]pubsub.BulkMessageEntry, 0)
+
+	for _, message := range msg.Entries {
+		entryID, err := strconv.Atoi(message.EntryID)
+		if err != nil {
+			log.Errorf("error sending message to dead letter as entry id not provided correctly, origin topic: %s dead letter topic %s err: %w", msg.Topic, deadLetterTopic, err)
+		}
+		if bulkResponses == nil || bulkResponses[entryID].Error != nil {
+			data = append(data, message)
+		}
+	}
+	req := &pubsub.BulkPublishRequest{
+		Entries:    data,
+		PubsubName: name,
+		Topic:      deadLetterTopic,
+		Metadata:   msg.Metadata,
+	}
+
+	err := a.BulkPublish(req)
+	if err != nil {
+		log.Errorf("error sending message to dead letter, origin topic: %s dead letter topic %s err: %w", msg.Topic, deadLetterTopic, err)
+	}
+	return err
+}
+
+func (a *DaprRuntime) bulkSubscribeTopic(ctx context.Context, policy resiliency.Runner,
+	name string, topic string, route TopicRouteElem,
+) error {
+	err := a.pubSubs[name].component.BulkSubscribe(ctx, pubsub.SubscribeRequest{
+		Topic:    topic,
+		Metadata: route.metadata,
+	}, func(ctx context.Context, msg *pubsub.BulkMessage) ([]pubsub.BulkSubscribeResponseEntry, error) {
```

Comment:
Is there a way this method can be "merged" into the handler above for the "regular" subscribe? Looks lilke a lot of code is duplicated, and then there's another method to maintain...

Reply:
Actually, this handler's input parameters, return types and working is totally different than normal subscribe handler. 
It definitely takes inspiration from the normal subscribe handler but treatment of incoming item(s) and thus the code is as well different.

Reply:
Ok